portal-web/docroot/html/taglib/ui/page_iterator/start.jsp

```diff
@@ -410,6 +412,81 @@ NumberFormat numberFormat = NumberFormat.getNumberInstance(locale);
 	}
 </script>
 
+<script data-senna-track="temporary" type="text/javascript">
+
+	var dropdown = document.getElementById("<%= namespace %>");
+	var button = dropdown.querySelector('.dropdown-toggle');
+	var list = dropdown.querySelector('.dropdown-menu');
+	var options = list.querySelectorAll('.dropdown-item');
+	var selectedItemValue = button.dataset.attribute;
+
+	function onButtonKeyDown(event) {
+		if (event.key === 'ArrowDown' || event.key === 'ArrowUp' || event.key === 'Enter' || event.key === 'Space') {
+			event.preventDefault();
+			list.classList.add('show');
+			button.setAttribute('aria-expanded', 'true');
+			var selectedOption = null;
+
+			Array.from(options).find(function(option) {
+				if (option.id == selectedItemValue) {
+					selectedOption = option;
+				}
+			});
+
+			if (selectedOption) {
+				selectedOption.focus();
+			}
+		}
+	}
+
+	button.addEventListener('keydown',onButtonKeyDown );
+
+	function handleKeyEvents(event, currIndex) {
+		if (event.key === 'ArrowDown') {
+			event.preventDefault();
+
+			if (currIndex < options.length - 1) {
+				options[currIndex + 1].focus();
+			}
+		} else if (event.key === 'ArrowUp') {
+			event.preventDefault();
+
+			if (currIndex > 0) {
+				options[currIndex - 1].focus();
+			}
+		}
+	}
+
+	options.forEach(function(option, index) {
+		option.addEventListener('keydown', function(event) {
```

Comment:
Ok I think we can improve this too I had forgotten to mention this in the other reviews, sorry about that. Instead of adding a listener for each option, we can just add a listener to the menu that works as a delegate and takes advantage of the DOM event bubbling.

For example:

```js
function handleKeyEvents(event, currIndex) {
		var currentIndex = options.indexOf(document.activeElement);

		if (event.key === 'ArrowDown') {
			event.preventDefault();
			if (currentIndex < options.length - 1) {
				options[currentIndex + 1].focus();
			}
		} else if (event.key === 'ArrowUp') {
			event.preventDefault();
			if (currentIndex > 0) {
				options[currentIndex - 1].focus();
			}
		}
	}

list.addEventListener('keydown', handleKeyEvents);
```

Reply:
Fixed